packages/pages/src/common/src/template/internal/getTemplateFilepaths.ts

```diff
@@ -0,0 +1,37 @@
+import glob from "glob";
+import path from "path";
+import { Path } from "../../project/path.js";
+
+/**
+ * Get all the template files in the provided template folder path(s).
+ * If there are two files that share the same name between the two
+ * provided template folder paths, only the file in domain template
+ * folder path is collected.
+ *
+ * @param templatesRoot  The folder path where all template files live
+ * @param templatesDomain The folder path where the template files of a specific domain live
+ * @returns a list of template filepaths
+ */
+export const getTemplateFilepaths = (
+  templatesRoot: Path,
+  templatesDomain?: Path
```

Comment:
Small nit: I think we can get rid of the `templates` prefix for both variables. It's implied by the method name.